File: soal3.java
import java.util.*;

class Crow{
    public String name;
    public int age;
    public String breed;
    
    public Crow(String nm, int ag, String br){
        this.name = nm;
        this.age =  ag;
        this.breed = br;
    }
    
    public String toString(){
        return "Crow{name = '" + this.name +"', age = "+ this.age + ", breed = '"+ this.breed +"'}";
    }
}
class Main {
    public static void main(String[] args) {
        List<Crow> test =  new ArrayList<>();
        test.add(new Crow("Belle",3,"Angus"));
        test.add(new Crow("Anna",2,"Holstein"));
        
        test.sort((a,b) -> {
            int nameComp = a.name.compareTo(b.name);
            if (nameComp != 0 ) return nameComp;
            
            int ageComp = Integer.compare(a.age,b.age);
            if (ageComp != 0 ) return ageComp;
            
            return a.breed.compareTo(b.breed);
            
        });
        
        for(Crow crow : test){
            System.out.println(crow);
        }
    }
}

File: soal-3.php
<?php

function sortCROW($test) {
    usort($test, function($a, $b) {
        if ($a['name'] < $b['name']) {
            return -1; 
        } elseif ($a['name'] > $b['name']) {
            return 1;
        } else {

          if ($a['age'] < $b['age']) {
                return -1;
            } elseif ($a['age'] > $b['age']) {
                return 1; 
            } else {
  
                if ($a['breed'] < $b['breed']) {
                    return -1; 
                } elseif ($a['breed'] > $b['breed']) {
                    return 1;
                } else {
                    return 0; 
                }
            }
        }
    });
    return $test;
}


$test = [
   ['name' => 'Belle', 'age' => 3, 'breed' => 'Angus'],
    ['name' => 'Anna', 'age' => 2, 'breed' => 'Holstein'],
];


$sortedCrows = sortCROW($test);
print_r($sortedCrows);

?>